src/libguac/guacamole/rwlock.h

```diff
@@ -42,15 +42,15 @@
  * An error code indicating that the calling thread is attempting to release a
  * lock that it does not control.
  */
-#define GUAC_REEANTRANT_LOCK_ERROR_DOUBLE_RELEASE 1
+#define GUAC_RWLOCK_ERROR_DOUBLE_RELEASE 1
 
 /**
  * The lock cannot be acquired because the lock has been already been
  * reentrantly acquired too many times, exhausting the capacity of this library
- * to track this lock. The lock must be released using guac_release_lock()
+ * to track this lock. The lock must be released using guac_rwlock_release_lock()
  * before it can be reacquired.
  */
-#define GUAC_REEANTRANT_LOCK_ERROR_TOO_MANY 2
+#define GUAC_RWLOCK_ERROR_TOO_MANY 2
```

Comment:
Rather than define new error codes and constants, I think it would be best to:

1. Use a non-zero return generically to represent that the operation failed.
2. Use `guac_error` (the libguac analogy to `errno`), `guac_error_message`, and the established `guac_status` constants (like `GUAC_STATUS_INVALID_ARGUMENT` and `GUAC_STATUS_TOO_MANY`) to make error details available.